kafka_consumer/datadog_checks/kafka_consumer/kafka_consumer.py

```diff
@@ -16,12 +27,448 @@ class KafkaCheck(AgentCheck):
+            api_version=self.instance.get('kafka_client_api_version'),
+            # While we check for SASL/SSL params, if not present they will default to the kafka-python values for
+            # plaintext connections
+            security_protocol=self.instance.get('security_protocol', 'PLAINTEXT'),
+            sasl_mechanism=self.instance.get('sasl_mechanism'),
+            sasl_plain_username=self.instance.get('sasl_plain_username'),
+            sasl_plain_password=self.instance.get('sasl_plain_password'),
+            sasl_kerberos_service_name=self.instance.get('sasl_kerberos_service_name', 'kafka'),
+            sasl_kerberos_domain_name=self.instance.get('sasl_kerberos_domain_name'),
+            ssl_cafile=self.instance.get('ssl_cafile'),
+            ssl_check_hostname=self.instance.get('ssl_check_hostname', True),
+            ssl_certfile=self.instance.get('ssl_certfile'),
+            ssl_keyfile=self.instance.get('ssl_keyfile'),
+            ssl_crlfile=self.instance.get('ssl_crlfile'),
+            ssl_password=self.instance.get('ssl_password'),
+        )
+        # Force initial population of the local cluster metadata cache
+        kafka_admin_client._client.poll(future=kafka_admin_client._client.cluster.request_update())
+        if kafka_admin_client._client.cluster.topics(exclude_internal_topics=False) is None:
+            raise RuntimeError("Local cluster metadata cache did not populate.")
+        return kafka_admin_client
+
+    def _get_highwater_offsets(self):
+        """Fetch highwater offsets for topic_partitions in the Kafka cluster.
+
+        Do this for all partitions in the cluster because even if it has no consumers, we may want to measure whether
+        producers are successfully producing.
+
+        If monitor_all_broker_highwatermarks is True, will fetch for all partitions in the cluster. Otherwise highwater
+        mark offsets will only be fetched for topic partitions where this check run has already fetched a consumer
+        offset.
+
+        Internal Kafka topics like __consumer_offsets, __transaction_state, etc are always excluded.
+
+        Any partitions that don't currently have a leader will be skipped.
+
+        Sends one OffsetRequest per broker to get offsets for all partitions where that broker is the leader:
+        https://cwiki.apache.org/confluence/display/KAFKA/A+Guide+To+The+Kafka+Protocol#AGuideToTheKafkaProtocol-OffsetAPI(AKAListOffset)
+
+        For speed, all the brokers are queried in parallel using callbacks. The callback flow is:
+            1. Issue an OffsetRequest to every broker
+            2. Attach a callback to each OffsetResponse that parses the response and saves the highwater offsets.
+        """
+        highwater_futures = []  # No need to store on object because the callbacks don't create additional futures
+
+        # No sense fetching highwatever offsets for internal topics
+        internal_topics = {
```

Comment:
Let's make this a constant

Reply:
Excellent idea. Added to legacy check here: https://github.com/DataDog/integrations-core/pull/4403